auth/sshlogin.go

```diff
@@ -129,11 +127,15 @@ func readSshKeyAndPassphrase(sshKeyPath, sshPassphrase string) ([]byte, []byte,
 }
 
 func IsEncrypted(buffer []byte) (bool, error) {
-	block, _ := pem.Decode(buffer)
-	if block == nil {
-		return false, errors.New("SSH: no key found")
+	_, err := ssh.ParsePrivateKey(buffer)
+	if err != nil {
+		if _, ok := err.(*ssh.PassphraseMissingError); ok {
+			// Key is encrypted
+			return true, nil
+		}
 	}
-	return strings.Contains(block.Headers["Proc-Type"], "ENCRYPTED"), nil
+	// Key is not encrypted or an error occurred
+	return false, nil
```

Comment:
Shouldn't we return the err here? We're always returning nil.

Reply:
Right